src/we-blog-init.1
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.32
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "WE-BLOG-INIT 1"
.TH WE-BLOG-INIT 1 "2012-07-12" "Version 0.9" "We-Blog Documentation"
.SH "NAME"
we\-blog\-init \- creates or recovers a We\-Blog repository
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBwe-blog-init\fR [\fB\-fqV\fR] [\fB\-b\fR \fIdirectory\fR]
.PP
\&\fBwe-blog-init\fR \fB\-h\fR|\fB\-v\fR
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBwe-blog-init\fR either creates a fresh new We-Blog repository, or
recovers an existing one in case it is corrupted. Optionally, it can also
revert a configuration and default templates to their original state,
leaving all user data (that is, both blog posts and pages) intact.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-b\fR \fIdirectory\fR, \fB\-\-blogdir\fR \fIdirectory\fR" 4
.IX Item "-b directory, --blogdir directory"
Allows you to specify a \fIdirectory\fR in which the We-Blog repository
is to be placed. The default option is a current working directory.
.IP "\fB\-f\fR, \fB\-\-force\fR" 4
.IX Item "-f, --force"
Reverts existing configuration, theme, and language files to their initial
state. By default, these files are kept intact.
.IP "\fB\-q\fR, \fB\-\-quiet\fR" 4
.IX Item "-q, --quiet"
Disables displaying of unnecessary messages.
.IP "\fB\-V\fR, \fB\-\-verbose\fR" 4
.IX Item "-V, --verbose"
Enables displaying of all messages, including a list of created files.
.IP "\fB\-h\fR, \fB\-\-help\fR" 4
.IX Item "-h, --help"
Displays usage information and exits.
.IP "\fB\-v\fR, \fB\-\-version\fR" 4
.IX Item "-v, --version"
Displays version information and exits.
.SH "EXAMPLE USAGE"
.IX Header "EXAMPLE USAGE"
Create a new blog in a current directory:
.PP
.Vb 2
\&        $ we-blog-init
\&        Created a We-Blog repository in .we-blog.
.Ve
.PP
Create a new blog in ~/public_html:
.PP
.Vb 2
\&        $ we-blog-init -b ~/public_html
\&        Created a We-Blog repository in /home/joe/public_html/.we-blog.
.Ve
.PP
Revert a configuration file and default templates to their initial state:
.PP
.Vb 2
\&        $ we-blog-init -f
\&        Recovered a We-Blog repository in .we-blog.
.Ve
.PP
Or if you want to see what files have been reverted:
.PP
.Vb 6
\&        $ we-blog-init -fV
\&        Created .we-blog/config
\&        Created .we-blog/theme/default.html
\&        Created .we-blog/style/default.css
\&        Created .we-blog/lang/en_US
\&        Recovered a We-Blog repository in .we-blog.
.Ve
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\fBwe-blog-config\fR(1), \fBwe-blog-add\fR(1)
.SH "BUGS"
.IX Header "BUGS"
To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/we\-blog/issues/>, or visit the
discussion group at <https://groups.google.com/d/forum/tonk\-we\-blog>.
.SH "COPYRIGHT"
.IX Header "COPYRIGHT"
Copyright (c) 2008\-2011 Jaromir Hradilek / 2011\-2012 Ton Kersten
.PP
This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but \s-1WITHOUT\s0 \s-1ANY\s0 \s-1WARRANTY\s0;
without even the implied warranty of \s-1MERCHANTABILITY\s0 or \s-1FITNESS\s0 \s-1FOR\s0 A
\&\s-1PARTICULAR\s0 \s-1PURPOSE\s0.


ChangeLog
2012-07-10  Ton Kersten <we-blog@tonkersten.com>
  * Declared 0.8 stable
  * Switched to branch 0.9 for further development

2012-07-09  Ton Kersten <we-blog@tonkersten.com>
  * Changed a lot of things, but they all boil down to
    bugfixes and code reshuffeling

2011-10-07  Ton Kersten <we-blog@tonkersten.com>
  * Added 'alt' text for pages to show on mouse-over.

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * And switched to branch 0.8 for further development

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * Increased we-blog to version 0.7
  * Added 'smiley' converter
  * Published the project on GitHub

2011-08-15  Ton Kersten <we-blog@tonkersten.com>
  * Forked BlazeBlogger 1.2.0 to we-blog 0.5
    This because development of Blaze seems to be stalled

2011-02-18  Jaromir Hradilek <jhradilek@gmail.com>

  * INSTALL, Makefile, README, src/we-blog-add.pl, src/we-blog-config.pl,
  src/we-blog-edit.pl, src/we-blog-init.pl, src/we-blog-list.pl, src/we-blog-log.pl,
  src/we-blog-make.pl, src/we-blog-remove.pl, unix/we-blog.sh, unix/man/man1/we-blog.1,
  windows/we-blog.bat: Increased the version number.

  * unix/man/man1/we-blog.1: Updated the manual page for the command wrapper.

# $Id: ChangeLog 4 2012-07-10 09:34:11 tonk $


src/we-blog-init.pl
#!/usr/bin/env perl
# vi: set sw=4 ts=4 ai:
# $Id: we-blog-init.pl 3 2012-07-12 10:25:52 tonk $

# we-blog-init - creates or recovers a We-Blog repository
# Copyright (c) 2011-2012 Ton Kersten
# Copyright (c) 2008-2011 Jaromir Hradilek

# This program is  free software:  you can redistribute it and/or modify it
# under  the terms  of the  GNU General Public License  as published by the
# Free Software Foundation, version 3 of the License.
#
# This program  is  distributed  in the hope  that it will  be useful,  but
# WITHOUT  ANY WARRANTY;  without  even the implied  warranty of MERCHANTA-
# BILITY  or  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
# License for more details.
#
# You should have received a copy of the  GNU General Public License  along
# with this program. If not, see <http://www.gnu.org/licenses/>.

use strict;
use warnings;
use File::Basename;
use File::Path;
use File::Spec::Functions;
use Getopt::Long;

# Set the library path and use our own module
use lib dirname($0);
use We;

# Display usage information:
sub display_help {
	# Display the usage:
	print << "END_HELP";
Usage: $NAME [-fqV] [-b DIRECTORY]
			 $NAME -h|-v

	-b, --blogdir DIRECTORY     specify a directory in which the We-Blog
	                            repository is to be placed
	-f, --force                 revert existing configuration, theme, and
	                            language files to their initial state
	-q, --quiet                 do not display unnecessary messages
	-V, --verbose               display all messages, including a list of
	                            created files
	-h, --help                  display this help and exit
	-v, --version               display version information and exit
END_HELP

	# Return success:
	return 1;
}

# Create the default configuration file:
sub create_conf {
	# Prepare the configuration file name:
	my $file = catfile($blogdir, $weblog, 'config');

	# Unless explicitly requested, do not overwrite the existing file:
	return 1 if (-e $file && !$force);

	# Open the file for writing:
	open(FILE, ">$file") or return 0;

	# Write the default configuration to the file:
	print FILE << 'END_CONFIG';
## This is the default We-Blog configuration file. The recommended way
## to set up your blog is to leave this file intact, and use we-blog-config
## instead. However, if you prefer to configure We-Blog by hand, read
## the instructions below, and replace the value on the right of the equal
## sign.

## General blog settings. Available options are:
##
##   title         A title of your blog.
##   subtitle      A subtitle of your blog.
##   description   A brief description of your blog.
##   keywords      A comma-separated list of keywords.
##   theme         A theme for your blog. It must point to an existing file
##                 in the .we-blog/theme/ directory.
##   style         A style sheet for your blog. It must point to an exis-
##                 ting file in the .we-blog/style/ directory.
##   lang          A translation of your blog. It must point to an existing
##                 file in the .we-blog/lang/ directory.
##   posts         A number of blog posts to be listed on a single page.
##
[blog]
title=Blog Title
subtitle=blog subtitle
description=blog description
keywords=blog keywords
theme=default.html
style=default.css
lang=en_US
posts=10
baseurl=http://blog.example.org

## Color settings. Available options are:
##
##   list          A boolean to enable (true) or disable (false) colors in
##                 the we-blog-list output.
##   log           A boolean to enable (true) or disable (false) colors in
##                 the we-blog-log output.
##
[color]
list=false
log=false

## Advanced We-Blog settings. Available options are:
##
##   doctype       A document type. It can be either html for HTML, or
##                 xhtml for the XHTML standard.
##   extension     A file extension.
##   encoding      A character encoding. It has to be in a form that is re-
##                 cognized by W3C standards.
##   editor        An external text editor. When supplied, this option
##                 overrides the system-wide settings.
##   processor     An external application to be used to process newly ad-
##                 ded or edited blog posts and pages. You must supply %in%
##                 and %out% in place of an input and output file name res-
##                 pectively.
##
[core]
doctype=html
extension=html
encoding=UTF-8
editor=
processor=

## RSS feed settings. Available options are:
##
##  baseurl        A URL of your blog, for example "http://example.com/".
##  posts          A number of blog posts to be listed in the feed.
##  fullposts      A boolean to enable (true) or disable (false) inclusion
##                 of the whole content of a blog post in the feed.
##
[feed]
baseurl=
posts=10
fullposts=false

## Social network settings. Available options are:
## (If not defined it will NOT be shown)
##
##  social        1 -> defined and a social "bar" is shown
##  twitter       Twitter id.         Results in: http://twitter.com/%twitter%
##  googleplus    Google Plus ID.     Results in: https://plus.google.com/%googleplus%/posts
##  linkedin      Linked In short ID. Results in: http://www.linkedin.com/in/%linkedin%
##  skype         Skype ID.           Results in: skype:%skype%?add
##  github        GitHub ID.          Results in: https://github.com/%github%
##
[social]
social=1
twitter=
googleplus=
linkedin=
skype=
github=

## Website statics settings. Available options are:
##
##  analytics     Google Analytics ID
##  piwik         PiWik baseurl (e.g. example.com/pi/piwik)
##  piwiksite     PiWik site id
##
[stats]
analytics=
piwik=
piwiksite=

## Advanced post settings. Available options are:
##
##  author         A location of a blog post author name. It can either be
##                 placed above the post (top), below it (bottom), or no-
##                 where on the page (none).
##  date           A location of a date of publishing. It can either be
##                 placed above the post (top), below it (bottom), or no-
##                 where on the page (none).
##  tags           A location of post tags. They can either be placed above
##                 the post (top), below it (bottom), or nowhere on the
##                 page (none).
##  showid         Show the post id with every post. Can be 'yes' of 'no'
##
[post]
author=top
date=top
tags=top
showid=no

## User settings. Available options are:
##
##   name          Your full name to be used in the copyright notice, and
##                 as the default post author.
##   nickname      Your nickname to be used as the default post author.
##                 When supplied, this option overrides the above setting.
##   email         Your email address.
##
[user]
name=admin
nickname=
email=admin@localhost

## Post appends
##
## These should be on a single line and all '%n%' will be replaced with a newline
##
##   index         Things to append for each post on the index page(s)
##   single        Things to append for each post when 'zoomed in'
[appends]
index=
single=

END_CONFIG

	# Close the file:
	close(FILE);

	# Report success:
	print "Created $file\n" if $verbose > 1;

	# Return success:
	return 1;
}

# Create the default theme file:
sub create_theme {
	# Prepare the theme file name:
	my $file = catfile($blogdir, $weblog, 'theme', 'default.html');

	# Unless explicitly requested, do not overwrite the existing file:
	return 1 if (-e $file && !$force);

	# Open the file for writing:
	open(FILE, ">$file") or return 0;

	# Write the default theme to the file:
	print FILE << 'END_THEME';
<!-- START-DOCUMENT -->
<head>
	<!-- content-type -->
	<!-- generator -->
	<!-- lastrun -->
	<!-- copyright -->
	<!-- date -->
	<!-- description -->
	<!-- keywords -->
	<!-- stylesheet -->
	<!-- feed -->
	<title><!-- page-title --></title>
</head>

<body>

<div id="wrapper">
	<div id="shadow">
		<div id="heading">
			<h1><a href="%home%" rel="index"><!-- title --></a></h1>
			<!-- subtitle -->
		</div>

		<div id="menu">
			<ul>
<li><a href="%home%" rel="index">Home</a></li>
<!-- pages -->
			</ul>
		</div>
	</div>

	<div id="content">
<!-- content -->
	</div>

	<div id="sidebar">
		<h2>Categories</h2>
		<ul>
<!-- tags -->
		</ul>

		<h2>Archive</h2>
		<ul>
<!-- archive -->
		</ul>

		<h2>Links</h2>
		<ul>
<li><a href="http://tonkersten.com/we-blog">We-Blog</a></li>
		</ul>
	</div>

	<div id="footer">
		Copyright &copy; <!-- year --> <!-- name -->.
		Powered by <a href="http://tonkersten.com/we-blog">We-Blog</a>.
	</div>
</div>

</body>
<!-- END-DOCUMENT -->
END_THEME

	# Close the file:
	close(FILE);

	# Report success:
	print "Created $file\n" if $verbose > 1;

	# Return success:
	return 1;
}

# Create the default style sheet:
sub create_style {
	# Prepare the style sheet file name:
	my $file = catfile($blogdir, $weblog, 'style', 'default.css');

	# Unless explicitly requested, do not overwrite the existing file:
	return 1 if (-e $file && !$force);

	# Open the file for writing:
	open(FILE, ">$file") or return 0;

	# Write the default style style sheet to the file:
	print FILE << 'END_STYLE';
/* default.css, the default theme for We-Blog
 *
 * Copyright (c) 2011-2012 Ton Kersten
 * Copyright (c) 2009-2010 Jaromir Hradilek
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTA-
 * BILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
 * License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

body {
	margin: 0px 0px 10px 0px;
	padding: 0px;
	color: #000000;
	background-color: #e7e7e7;
	font-family: "DejaVu Sans", Arial, sans;
	font-size: small;
}

#wrapper {
	margin: auto;
	padding: 0px;
	width: 768px;
	border-left: 1px solid #d6d6d6;
	border-right: 1px solid #d6d6d6;
	border-bottom: 1px solid #d6d6d6;
	background-color: #ffffff;
}

#shadow {
	margin: 0px;
	padding: 0px;
	border-bottom: 2px solid #e7e7e7;
}

#heading {
	width: 728px;
	padding: 20px;
	background-color: #2e2e2e;
	border-bottom: 2px solid #2a2a2a;
	border-top: 2px solid #323232;
	color: #d0d0d0;
}

#heading a, #heading h1 {
	margin: 0px;
	text-decoration: none;
	color: #ffffff;
}

#heading a:hover {
	text-decoration: underline;
}

#menu {
	width: 768px;
	border-top: 1px solid #5f5f5f;
	border-bottom: 1px solid #3d3d3d;
	background-color: #4e4e4e;
}

#menu ul {
	padding: 4px 15px 4px 15px;
	margin: 0px;
	list-style-type: none;
}

#menu li {
	display: inline;
	padding: 4px 10px 5px 10px;
	margin: 0px;
}

#menu li:hover {
	background-color: #3d3d3d;
	border-bottom: 2px solid #dddddd;
	border-top: 1px solid #4e4e4e;
}

#menu a {
	color: #ffffff;
	font-size: x-small;
	text-decoration: none;
}

#menu a:hover {
	text-decoration: underline;
}

#content {
	float: left;
	margin: 0px;
	padding: 10px 10px 20px 20px;
	width: 528px;
	text-align: justify;
}

#content h2.post {
	margin-bottom: 0px;
	padding-bottom: 0px;
}

#content .post a {
	text-decoration: none;
	color: #9acd32;
}

#content a {
	text-decoration: none;
	color: #4e9a06;
}

#content a:hover {
	text-decoration: underline;
}

#content .information {
	font-size: x-small;
	color: #4e4e4e;
}

#content .information a {
	color: #4e9a06;
	text-decoration: underline;
}

#content .information a:hover {
	text-decoration: none;
}

#content .post-heading {
	font-size: x-small;
	color: #4e4e4e;
}

#content .post-heading a {
	color: #4e9a06;
	text-decoration: underline;
}

#content .post-heading a:hover {
	text-decoration: none;
}

#content .post-footer {
	font-size: x-small;
	color: #4e4e4e;
	padding: 4px 2px 4px 2px;
	border-top: 1px solid #e7e7e7;
	border-bottom: 1px solid #e7e7e7;
	background-color: #f8f8f8;
}

#content .post-footer a {
	color: #4e9a06;
	text-decoration: underline;
}

#content .post-footer a:hover {
	text-decoration: none;
}

#content .section {
	text-align: right;
	font-size: x-small;
	color: #808080;
}

#content .previous {
	padding: 10px 0px 10px 0px;
	float: left;
}

#content .next {
	padding: 10px 0px 10px 0px;
	float: right;
}

#sidebar {
	float: right;
	margin: 0px;
	padding: 10px 20px 20px 0px;
	width: 180px;
}

#sidebar h2 {
	font-size: small;
}

#sidebar ul {
	list-style-type: none;
	padding-left: 1em;
	margin-left: 0px;
}

#sidebar a {
	text-decoration: underline;
	color: #4e9a06;
}

#sidebar a:hover {
	text-decoration: none;
}

#footer {
	clear: both;
	margin: 0px;
	padding: 10px 20px 10px 20px;
	border-top: 2px solid #e7e7e7;
	border-bottom: 1px solid #3d3d3d;
	background-color: #4e4e4e;
	text-align: right;
	font-size: x-small;
	color: #d0d0d0;
}

#footer a {
	color: #ffffff;
	text-decoration: none;
}

#footer a:hover {
	text-decoration: underline;
}
END_STYLE

	# Close the file:
	close(FILE);

	# Report success:
	print "Created $file\n" if $verbose > 1;

	# Return success:
	return 1;
}

# Create the default localization file:
sub create_lang {
	# Prepare the localization file name:
	my $file = catfile($blogdir, $weblog, 'lang', 'en_US');

	# Unless explicitly requested, do not overwrite the existing file:
	return 1 if (-e $file && !$force);

	# Open the file for writing:
	open(FILE, ">$file") or return 0;

	# Write the default localization to the file:
	print FILE << 'END_LANG';
[lang]
archive=Archive for
tags=Posts tagged as
taglist=List of tags
previous=&laquo; Previous
next=Next &raquo;
more=Read more &raquo;
postedon=
postedby=by
taggedas=tagged as
january=January
february=February
march=March
april=April
may=May
june=June
july=July
august=August
september=September
october=October
november=November
december=December
END_LANG

	# Close the file:
	close(FILE);

	# Report success:
	print "Created $file\n" if $verbose > 1;

	# Return success:
	return 1;
}

# Set up the option parser:
Getopt::Long::Configure('no_auto_abbrev', 'no_ignore_case', 'bundling');

# Process command line options:
GetOptions(
	'help|h'        => sub { display_help();    exit 0; },
	'version|v'     => sub { display_version(); exit 0; },
	'force|f'       => sub { $force   = 1;     },
	'quiet|q'       => sub { $verbose = 0;     },
	'verbose|V'     => sub { $verbose = 2;     },
	'blogdir|b=s'   => sub { $blogdir = $_[1]; },
);

# Detect superfluous options:
exit_with_error("Invalid option `$ARGV[0]'.", 22) if (scalar(@ARGV) != 0);

# Decide whether to create or recover the repository:
my $action = (-d catdir($blogdir, )) ? 'Recovered' : 'Created';

# Create the directory tree:
eval {
	mkpath(
		[
			catdir($blogdir, $weblog, 'lang'),
			catdir($blogdir, $weblog, 'theme'),
			catdir($blogdir, $weblog, 'style'),
			catdir($blogdir, $weblog, 'pages', 'head'),
			catdir($blogdir, $weblog, 'pages', 'body'),
			catdir($blogdir, $weblog, 'pages', 'raw'),
			catdir($blogdir, $weblog, 'posts', 'head'),
			catdir($blogdir, $weblog, 'posts', 'body'),
			catdir($blogdir, $weblog, 'posts', 'raw'),
		],
		0 # Do not be verbose.
	);
};

# Make sure the directory tree creation was successful:
exit_with_error("Creating directory tree: $@", 13) if $@;

# Create the default configuration file:
create_conf()
	or display_warning("Unable to create the default configuration file.");

# Create the default theme:
create_theme()
	or display_warning("Unable to create the default theme.");

# Create the default style sheet:
create_style()
	or display_warning("Unable to create the default style sheet.");

# Create the default localization:
create_lang()
	or display_warning("Unable to create the default localization.");

# Create the default log file:
add_to_log("$action a We-Blog repository.")
	or display_warning("Unable to log the event.");

# Report success:
print "$action a We-Blog repository in " .
			catdir($blogdir, ) . ".\n" if $verbose;

# Return success:
exit 0;

__END__

=head1 NAME

we-blog-init - creates or recovers a We-Blog repository

=head1 SYNOPSIS

B<we-blog-init> [B<-fqV>] [B<-b> I<directory>]

B<we-blog-init> B<-h>|B<-v>

=head1 DESCRIPTION

B<we-blog-init> either creates a fresh new We-Blog repository, or
recovers an existing one in case it is corrupted. Optionally, it can also
revert a configuration and default templates to their original state,
leaving all user data (that is, both blog posts and pages) intact.

=head1 OPTIONS

=over

=item B<-b> I<directory>, B<--blogdir> I<directory>

Allows you to specify a I<directory> in which the We-Blog repository
is to be placed. The default option is a current working directory.

=item B<-f>, B<--force>

Reverts existing configuration, theme, and language files to their initial
state. By default, these files are kept intact.

=item B<-q>, B<--quiet>

Disables displaying of unnecessary messages.

=item B<-V>, B<--verbose>

Enables displaying of all messages, including a list of created files.

=item B<-h>, B<--help>

Displays usage information and exits.

=item B<-v>, B<--version>

Displays version information and exits.

=back

=head1 EXAMPLE USAGE

Create a new blog in a current directory:

	$ we-blog-init
	Created a We-Blog repository in .we-blog.

Create a new blog in ~/public_html:

	$ we-blog-init -b ~/public_html
	Created a We-Blog repository in /home/joe/public_html/.we-blog.

Revert a configuration file and default templates to their initial state:

	$ we-blog-init -f
	Recovered a We-Blog repository in .we-blog.

Or if you want to see what files have been reverted:

	$ we-blog-init -fV
	Created .we-blog/config
	Created .we-blog/theme/default.html
	Created .we-blog/style/default.css
	Created .we-blog/lang/en_US
	Recovered a We-Blog repository in .we-blog.

=head1 SEE ALSO

B<we-blog-config>(1), B<we-blog-add>(1)

=head1 BUGS

To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/we-blog/issues/>, or visit the
discussion group at <https://groups.google.com/d/forum/tonk-we-blog>.

=head1 COPYRIGHT

Copyright (c) 2008-2011 Jaromir Hradilek / 2011-2012 Ton Kersten

This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

=cut
